### PR TITLE
Feat(eos_cli_config_gen): Option to set RCF route bgp address-family vpn-ipv6/ipv4 and redistribute.

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Adding RCF for router-bgp address-family vpn-ipv4, vpn-ipv6 and route redistribute.

## Related Issue(s)

part of #2397 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 1]
```diff
@@ -1780,11 +1789,32 @@ keys:
...
+                  description: Outbound rcf function name without parenthesis.
           neighbor_default_encapsulation_mpls_next_hop_self:
             type: dict
             keys:
               source_interface:
                 type: str
+          redistribute_routes:
+            type: list
+            primary_key: source_protocol
+            convert_types:
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: you can remove `convert_types` here

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 2]
```diff
@@ -1834,11 +1870,32 @@ keys:
...
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: same as above

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md [hunk 3]
```diff
@@ -1574,8 +1600,20 @@
...
+            # Outbound rcf function name without parenthesis.
+            rcf_out: <str>
         neighbor_default_encapsulation_mpls_next_hop_self:
           source_interface: <str>
+        redistribute_routes:
+          - source_protocol: <str; required; unique>
+            include_leaked: <bool>
+
+            # RCF function name without parenthesis.
+            rcf: <str>
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md]: One thing i noticed that rcf for redistribute is supported for some of the options not on all. Can we update the description accordingly.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md]: Added description.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md [hunk 4]
```diff
@@ -1574,8 +1604,24 @@
...
+            # Inbound RCF function name with parenthesis.
+            # Example: MyFunction(myarg).
+            rcf_in: <str>
+
+            # Outbound RCF function name with parenthesis.
+            # Example: MyFunction(myarg).
+            rcf_out: <str>
         neighbor_default_encapsulation_mpls_next_hop_self:
           source_interface: <str>
+        redistribute_routes:
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md]: I dont think we need this as applying redistribute route under address family will redirect to global config thats the reason its not added for route map also.
cc: @ClausHolbechArista

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md]: I think it depends on the specific routing protocols and capabilities, but for now I think we can just remove this and wait for a specific request.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/router-bgp.md]: Removed redistribute_routes from vpn-ipv4/ipv6

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-bgp-base.cfg [hunk 5]
```diff
@@ -91,7 +91,9 @@ router bgp 65101
    aggregate-address 1.12.1.0/24 as-set advertise-map ADV-MAP supress-map SUP-MAP summary-only attribute-map RM-ATTRIBUTE match-map RM-MATCH advertise-only
    aggregate-address 2.2.1.0/24
    redistribute bgp leaked route-map RM-REDISTRIBUTE-BGP
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-bgp-base.cfg:93]: This bgp leaked config reflects at the end. Need to check. Output generated from AVD
```
   redistribute bgp leaked route-map RM-REDISTRIBUTE-BGP
   redistribute connected route-map RM-CONN-2-BGP
   redistribute dynamic route-map Route_map_dynamic
   redistribute ospf include leaked
   redistribute static rcf Router_BGP_Static()
```

output from EOS
```
   redistribute connected route-map RM-CONN-2-BGP
   redistribute ospf include leaked
   redistribute static rcf Router_BGP_Static()
   redistribute dynamic route-map Route_map_dynamic
   redistribute bgp leaked route-map RM-REDISTRIBUTE-BGP
```

Seems like EOS is ordering the protocol randomly in config.

**Review by MaheshGSLAB — APPROVED** [review]
Have one comment but i dont think that we can fix it now. Tested on lab LGTM

**Review by Shivani-gslab — APPROVED** [review]
Tested on EOS CLI.
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**